pyccel/ast/core.py

```diff
@@ -3303,6 +3316,10 @@ def functions(self):
     def hide(self):
         return self.functions[0].hide or self._args[2]
 
+    @property
+    def is_argument(self):
+        return self._args[3]
```

Comment:
Can we use this PR to make Instance use variables instead of storing everything in a big tuple? The more things there are in `self._args` the harder it gets to read the code

Reply:
Yes sure